include/navigation/differencers.h

```diff
@@ -136,6 +137,14 @@ class RMSDiff
     };
 };
 
+template<class T>
+bool allSame(cv::InputArray &arr)
+{
+    const cv::Mat_<T> m = arr.getMat();
+    return std::all_of(m.begin() + 1, m.end(),
+                       [&m](T val) { return val == *m.begin(); });
+}
```

Comment:
Is it worth trying to use e.g. an OpenCV histogram for this? They tend to have more hard-core optimizations than the normal C++ compiler will attempt.

Reply:
That's true. How would calculating a histogram instead work though? I feel I'm missing something